modules/visualization/dev-instance/stack.py

```diff
@@ -0,0 +1,180 @@
+        Tags.of(scope=cast(IConstruct, self)).add(key="DeploymentName", value=f"addf-{deployment_name}")
+
+        region = Stack.of(self).region
+        dep_mod = f"addf-{deployment_name}-{module_name}"
+
+        ####################################################################################################
+        # VPC
+        ####################################################################################################
+        vpc = Vpc.from_lookup(self, "VPC", vpc_id=vpc_id)
+
+        if ami_id is not None:
+            ami = MachineImage.lookup(name=ami_id)
+        else:
+            ami = MachineImage.from_ssm_parameter(default_ami_ssm_parameter_name, os=OperatingSystemType.LINUX)
+
+        instance_sg = SecurityGroup(
+            self,
+            id="instance-sg",
+            security_group_name=f"{module_name}-dev-instance-sg",
+            allow_all_outbound=True,
+            vpc=vpc,
+        )
+        instance_sg.add_ingress_rule(
+            Peer.any_ipv4(),
+            Port.tcp(8443),
+            "allow 8443 access everywhere",
+        )
+
+        with open(os.path.join("user-data", "script.sh"), "r") as f:
+            user_data_script = f.read()
+
+        dev_instance_role = aws_iam.Role(
+            self,
+            "dev-instance-role",
+            assumed_by=aws_iam.ServicePrincipal("ec2.amazonaws.com"),
+            managed_policies=[aws_iam.ManagedPolicy.from_aws_managed_policy_name("AmazonSSMManagedInstanceCore")],
+        )
+
+        cloudformation_policy_json = {
+            "Effect": "Allow",
+            "Action": ["cloudformation:DescribeStacks"],
+            "Resource": [Stack.of(self).stack_id],
+        }
+
+        dcv_license_policy_json = {
+            "Effect": "Allow",
+            "Action": ["s3:GetObject"],
+            "Resource": [f"arn:aws:s3:::dcv-license.{region}/*"],
+        }
+
+        s3_policy_json = {
+            "Effect": "Allow",
+            "Action": ["s3:Get*", "s3:List*", "s3:PutObject*", "s3:DeleteObject*"],
+            "Resource": ["arn:aws:s3:::addf*", "arn:aws:s3:::addf*/*"],
+        }
+
+        if s3_dataset_bucket:
+            public_aev_dataset_policy_json = {
+                "Effect": "Allow",
+                "Action": ["s3:Get*", "s3:List*"],
+                "Resource": [f"arn:aws:s3:::{s3_dataset_bucket}", f"arn:aws:s3:::{s3_dataset_bucket}/*"],
+            }
+
+        # Add the policies to the role
+        dev_instance_role.add_to_principal_policy(aws_iam.PolicyStatement.from_json(cloudformation_policy_json))
+        dev_instance_role.add_to_principal_policy(aws_iam.PolicyStatement.from_json(dcv_license_policy_json))
+        dev_instance_role.add_to_principal_policy(aws_iam.PolicyStatement.from_json(s3_policy_json))
+        if s3_dataset_bucket:
+            dev_instance_role.add_to_principal_policy(aws_iam.PolicyStatement.from_json(public_aev_dataset_policy_json))
+
+        for idx in range(0, instance_count):
```

Comment:
This is where the instance count is used

Reply:
I agree, but can we do this after reinvent?